src/Debug/Profile.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Debug;
+
+class Profile
+{
+    private string $id;
+
+    private ?string $parent_id;
+
+    private string $start_time;
+
+    /**
+     * @var bool If true, the profile is loaded from an old request so current debug info should not be accessible from this profile
+     */
+    private bool $is_readonly = false;
+
+    /**
+     * @var array|null Debug info for this profile. This is only set for loaded/readonly profiles.
+     */
+    private ?array $debug_info = null;
+
+    private array $additional_info = [];
+
+    private static ?self $current = null;
+
+    /**
+     * @var bool If true, top-level profiles will be saved to disk.
+     * This is false by default since these are usually initial pages loads and the data is directly used, so saving it is not useful.
+     */
+    private const SAVE_TOP_LEVEL_PROFILES = false;
+
+    /**
+     * The threshold for the size of the debug info before it is saved to disk rather than sent directly to the browser.
+     */
+    private const DEBUG_INFO_HEADER_THRESHOLD = 4096;
```

Comment:
Maybe session is a more appropriate place for this kind of data. Only problem would to prevent profiler session data to be displayed in profiler.

Reply:
That would probably be better as long as the profiles are removed as soon as possible to keep the session from getting too large and handle alternative storage like Redis. The one downside I see is that the requests to `ajax/debug.php` have to be blocking because it needs the session to stay open.

I removed the file saving and replaced it with a new `debug_profiles` array in the session. When getting the debug info, this property is excluded to avoid recursion. Previously, the session data wasn't saved for AJAX requests because of the large size, but that isn't an issue now that it isn't writing files so I re-enabled that.